### PR TITLE
controllers: create vrc sent by the provider

### PR DESCRIPTION
**Pull request by rewantsoni**

### CODE REVIEW

File: internal/controller/operatorconfigmap_controller.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/go-logr/logr"
```
Comment (leelavg) [inline internal/controller/operatorconfigmap_controller.go:20]: wrong placement, pls recheck.

File: internal/controller/operatorconfigmap_controller.go [hunk 2]
```diff
@@ -80,6 +80,7 @@ type OperatorConfigMapReconciler struct {
 	OperatorNamespace string
 	ConsolePort       int32
 	Scheme            *runtime.Scheme
+	AvailableCrds     map[string]bool
```
Comment (leelavg) [inline internal/controller/operatorconfigmap_controller.go:83]: what is using this feature?

File: internal/controller/storageclaim_controller.go [hunk 3]
```diff
@@ -115,20 +111,8 @@ func (r *StorageClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-				utils.NamePredicate(drClusterConfigCRDName),
-				utils.CrdCreateAndDeletePredicate(&r.log, drClusterConfigCRDName, r.AvailableCrds[drClusterConfigCRDName]),
-			),
-			builder.OnlyMetadata,
-		)
-
-	if r.AvailableCrds[drClusterConfigCRDName] {
-		bldr = bldr.Owns(&ramenv1alpha1.DRClusterConfig{}, builder.WithPredicates(generationChangePredicate))
-	}
+		Owns(&replicationv1alpha1.VolumeReplicationClass{}, builder.WithPredicates(generationChangePredicate)).
```
Comment (leelavg) [inline internal/controller/storageclaim_controller.go:114]: does this CRD always exist?

Comment (rewantsoni) [reply internal/controller/storageclaim_controller.go:114]: Yes, it is brought by csi-addons

File: internal/controller/operatorconfigmap_controller.go (hunk 2 above)
Comment (rewantsoni) [reply internal/controller/operatorconfigmap_controller.go:83]: removed it

File: internal/controller/operatorconfigmap_controller.go (hunk 1 above)
Comment (rewantsoni) [reply internal/controller/operatorconfigmap_controller.go]: done

**Pull request merged**